icu4c/source/common/cjmlbe.cpp

```diff
@@ -0,0 +1,513 @@
+        if (U_FAILURE(status)) {
+            return;
+        }
+    }
+    if (w4 != INVALID) {
+        arr[0] = w4;
+        length = 1;
+        featureList.adoptElement(concatChar(u"UW4:", arr, length, status), status);
+        if (U_FAILURE(status)) {
+            return;
+        }
+    }
+    if (w5 != INVALID) {
+        arr[0] = w5;
+        length = 1;
+        featureList.adoptElement(concatChar(u"UW5:", arr, length, status), status);
+        if (U_FAILURE(status)) {
+            return;
+        }
+    }
+    if (w6 != INVALID) {
+        arr[0] = w6;
+        length = 1;
+        featureList.adoptElement(concatChar(u"UW6:", arr, length, status), status);
+        if (U_FAILURE(status)) {
+            return;
+        }
+    }
+    if (w2 != INVALID && w3 != INVALID) {
+        arr[0] = w2;
+        arr[1] = w3;
+        length = 2;
+        featureList.adoptElement(concatChar(u"BW1:", arr, length, status), status);
+        if (U_FAILURE(status)) {
+            return;
+        }
+    }
+    if (w3 != INVALID && w4 != INVALID) {
+        arr[0] = w3;
+        arr[1] = w4;
+        length = 2;
+        featureList.adoptElement(concatChar(u"BW2:", arr, length, status), status);
+        if (U_FAILURE(status)) {
+            return;
+        }
+    }
+    if (w4 != INVALID && w5 != INVALID) {
+        arr[0] = w4;
+        arr[1] = w5;
+        length = 2;
+        featureList.adoptElement(concatChar(u"BW3:", arr, length, status), status);
+        if (U_FAILURE(status)) {
+            return;
+        }
+    }
+    if (w1 != INVALID && w2 != INVALID && w3 != INVALID) {
+        arr[0] = w1;
+        arr[1] = w2;
+        arr[2] = w3;
+        length = 3;
+        featureList.adoptElement(concatChar(u"TW1:", arr, length, status), status);
+        if (U_FAILURE(status)) {
+            return;
+        }
+    }
+    if (w2 != INVALID && w3 != INVALID && w4 != INVALID) {
+        arr[0] = w2;
+        arr[1] = w3;
+        arr[2] = w4;
+        length = 3;
+        featureList.adoptElement(concatChar(u"TW2:", arr, length, status), status);
+        if (U_FAILURE(status)) {
+            return;
+        }
+    }
+    if (w3 != INVALID && w4 != INVALID && w5 != INVALID) {
+        arr[0] = w3;
+        arr[1] = w4;
+        arr[2] = w5;
+        length = 3;
+        featureList.adoptElement(concatChar(u"TW3:", arr, length, status), status);
+        if (U_FAILURE(status)) {
+            return;
+        }
+    }
+    if (w4 != INVALID && w5 != INVALID && w6 != INVALID) {
+        arr[0] = w4;
+        arr[1] = w5;
+        arr[2] = w6;
+        length = 3;
+        featureList.adoptElement(concatChar(u"TW4:", arr, length, status), status);
+        if (U_FAILURE(status)) {
+            return;
+        }
+    }
+    if (isValid(b1)) {
+        featureList.adoptElement(cloneString(UnicodeString(u"UB1:").append(b1), status), status);
+        if (U_FAILURE(status)) {
```

Comment:
you do not need to check the status here after the call

Reply:
Done